### ISSUE TITLE
[BUG] "java.lang.IllegalArgumentException: -84 is not a valid id"

### ISSUE THREAD

**Issue by uranru** [opensearch-project/security#4521]
### Describe the bug

I use logstash to load data in Openserach.
After updating the version (2.13 -> 2.15) on several nodes, I began to receive this error.

### Related component

Other

### To Reproduce

Logstash (logstash:8.13.1) sends data to Opensearch.

### Expected behavior

No errors when loading data

### Additional Details

**Plugins**
Please list all plugins currently enabled.

**Screenshots**

**Host/Environment (please complete the following information):**

**Additional context**

**Comment by DarshitChanpura** [comment]
Related to: https://github.com/opensearch-project/security/issues/4494

**Comment by shikharj05** [comment]
In 2.11, custom serialization was introduced with https://github.com/opensearch-project/security/pull/2802

In 2.14, custom serialization was disabled and the plugin moved back to JDK serialization with https://github.com/opensearch-project/security/pull/4264

Hence, I think for a cluster upgrading from 2.11/2.12/2.13 to any 2.14+ version, the cluster can run into this issue due to different serialization methods. 

In 2.11/2.12/2.13 - serialization method is decided using [version check]( https://github.com/opensearch-project/security/blob/2.13/src/main/java/org/opensearch/security/transport/SecurityInterceptor.java#L153)

In 2.14+ - serialization is decided using version check in  [SerializationFormat](https://github.com/opensearch-project/security/blob/2.14/src/main/java/org/opensearch/security/support/SerializationFormat.java)

Potential fix: We need to consider if an existing cluster with custom serialization is being upgraded to a cluster that will use JDK serialization and change serialization method accordingly.

**Comment by scrawfor99** [comment]
[Triage] Hi @uranru thanks for filing this issue. I am going to close this issue in favor of https://github.com/opensearch-project/security/issues/4494 which is a bit older with some additional context. Thank you for filing this though as it is a problem.

**Comment by cwperks** [comment]
> Hence, I think for a cluster upgrading from 2.11/2.12/2.13 to any 2.14+ version, the cluster can run into this issue due to different serialization methods.

How does this happen though?

The serialization/de-serialization log requires the sending and receiving node to get the version of the opposite node on the channel correctly.

i.e.

Transmitting Node --------------------- Transport Channel ------------------------> Receiving Node

The Transmitting node needs to get the version of the receiving node to determine serialization method. Likewise, the receiving node needs to get the version of the transmitting node to determine how to deserialize.

This issue occurs when either the transmitting node or receiving node fails to get the accurate version of the opposite node.

This issue a red herring for improper delegation and it needs to be determined where the improper delegation is occurring.

When this issue was seen in < 2.11 nodes mixed with 2.11 nodes it was because Performance Analyzer was not properly delegating the `getVersion` implementation to the wrapped channel: https://github.com/opensearch-project/performance-analyzer/pull/609

**Comment by shikharj05** [comment]
> How does this happen though?

In this case, the version of the outer channel is not being checked [here](https://github.com/opensearch-project/security/blob/b61252947caa5b63e73c97b855b8a26d40422cd7/src/main/java/org/opensearch/security/ssl/transport/SecuritySSLRequestHandler.java#L91). Inner channel wrapped by PA doesn't implement the ``getVersion`` method, which would mean a cluster with PA would fallback to current version instead of version of transmitting node.

**Comment by DarshitChanpura** [comment]
TLDR from above, would implementing a version check in both channels resolve the issue?

**Comment by cwperks** [comment]
> In this case, the version of the outer channel is not being checked [here](https://github.com/opensearch-project/security/blob/b61252947caa5b63e73c97b855b8a26d40422cd7/src/main/java/org/opensearch/security/ssl/transport/SecuritySSLRequestHandler.java#L91). Inner channel wrapped by PA doesn't implement the getVersion method, which would mean a cluster with PA would fallback to current version instead of version of transmitting node.

What are the concrete classes of the outer channel and inner channel that produces the error submitted by OP?
